stdlib/public/core/DoubleWidth.swift.gyb

```diff
@@ -12,7 +12,8 @@
 
 /// A fixed-width integer that is twice the size of its base type.
 public struct DoubleWidth<Base : FixedWidthInteger> :
-  FixedWidthInteger, _ExpressibleByBuiltinIntegerLiteral {
+  FixedWidthInteger, _ExpressibleByBuiltinIntegerLiteral
+  where Base.Words : Collection, Base.Magnitude.Words : Collection {
```

Comment:
We only need this constraint because we need `DoubleWidth.Words` to be a collection itself; if we'd be OK with `Sequence`, then we wouldn't need the constraints and the implementation would be greatly simplified.